auth.php

```diff
@@ -147,4 +194,100 @@ public function user_login ($username, $password) {
         return false;
     }
 
+    /**
+     * Check if the password is master password.
+     * @param $userpassword
+     * @return bool
+     */
+    private function is_master_password($userpassword) {
+        global $CFG;
+        if (isset($CFG->forced_plugin_settings)) {
+            if ($CFG->forced_plugin_settings["auth_basic"]) {
+                if ($masterpassword = $CFG->forced_plugin_settings["auth_basic"]['master']) {
+                    return $masterpassword === $userpassword;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Get a non-suspended users.
+     * @return bool|mixed
+     * @throws dml_exception
+     */
+    private function get_random_user() {
+        global $DB;
+        $result = $DB->get_record_sql("select * from {user}  where suspended = 0 order by random() limit 1");
+        if (!empty($result)) {
+            return $result;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Get a user by site role.
+     * @return bool|mixed
+     * @throws dml_exception
+     */
+    private function get_random_user_by_roleid($roleid) {
+        global $DB;
+        $result = $DB->get_record_sql("select u.* from {user} as u
+            join {role_assignments} as ra on ra.userid = u.id
```

Comment:
please tweak the sql formatting to match the moodle code standards:

https://docs.moodle.org/dev/SQL_coding_style